dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2651,345 @@ void FileSprayer::pushParts()
+            blobCopyOp = std::make_unique<StartBlobCopyOperation>(std::move(blobClient->StartCopyFromUri(source)));
+            status = ApiCopyStatus::Pending;
+        }
+        catch (const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("AzureBlobClient startCopyFromURI failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+            throw MakeStringException(DFTERR_CopyFailed, "%s", e.ReasonPhrase.c_str());
+        }
+        return status;
+    }
+    ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength)
+    {
+        if (status!=ApiCopyStatus::Pending)
+            return status;
+        try
+        {
+            blobCopyOp->Poll();
+            Models::BlobProperties props = blobCopyOp->Value();
+            dateTime.setString(props.LastModified.ToString().c_str());
+            outputLength = props.BlobSize;
+            Blobs::Models::CopyStatus tstatus = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Blobs::Models::CopyStatus::Pending);
+            if (tstatus==Blobs::Models::CopyStatus::Success) // FYI. CopyStatus is an object so can't use switch statement
+                status = ApiCopyStatus::Success;
+            else if (tstatus==Blobs::Models::CopyStatus::Pending)
+                status = ApiCopyStatus::Pending;
+            else if (tstatus==Blobs::Models::CopyStatus::Aborted)
+                status = ApiCopyStatus::Aborted;
+            else
+                status = ApiCopyStatus::Failed;
+        }
+        catch(const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("Transfer using API .Poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            return ApiCopyStatus::Failed;
+        }
+        return status;
+    }
+    ApiCopyStatus abortCopy()
+    {
+        if (status==ApiCopyStatus::Aborted || status==ApiCopyStatus::Failed)
+            return status;
+        int64_t outputLength;
+        CDateTime dateTime;
+        status = getProgress(dateTime, outputLength);
+        switch (status)
+        {
+            case ApiCopyStatus::Pending:
+                try
+                {
+                    if (blobCopyOp->HasValue() && blobCopyOp->Value().CopyId.HasValue())
+                        blobClient->AbortCopyFromUri(blobCopyOp->Value().CopyId.Value().c_str());
+                    else // not sure that this can ever happen
+                        IERRLOG("Transfer using API .AbortCopy() failed: unexpected response from API");
+                    status = ApiCopyStatus::Aborted;
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("Abort copy operation failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+                    status = ApiCopyStatus::Failed;
+                }
+                break;
```

Comment:
Same comment about needing to delete if aborted.  (Or if the abort was too late?)